cpu/stm32f1/periph/adc.c

```diff
@@ -57,10 +57,28 @@ static inline void prep(adc_t line)
 {
     mutex_lock(&locks[adc_config[line].dev]);
     periph_clk_en(APB2, (RCC_APB2ENR_ADC1EN << adc_config[line].dev));
+
+    /* enable the ADC module */
+    dev(line)->CR2 |= ADC_CR2_ADON;
+
+    /* check if this channel is an internal ADC channel, if so
+     * enable the internal temperature and Vref */
+    if (adc_config[line].chan == 16 || adc_config[line].chan == 17) {
+        dev(line)->CR2 |= ADC_CR2_TSVREFE;
+    }
 }
 
 static inline void done(adc_t line)
 {
+    /* check if this channel is an internal ADC channel, if so
+     * disable the internal temperature and Vref */
+    if (adc_config[line].chan == 16 || adc_config[line].chan == 17) {
+        dev(line)->CR2 &= ~ADC_CR2_TSVREFE;
```

Comment:
I think this operation is idempotent and not harmful in case the channel was not internal. So just disabling it unconditionally should be safe to do and saves a bit of ROM.

Also: Clearing this bit in the reg should be at most 3 instructions (read CR2 into a registers, clear the bit, write the register back to CR2). I noticed that the STM32F1 performs quite poorly when branching (I guess the pipeline is flushed, but I didn't care to check this). So this might just also turn out being faster (for both internal and regular ADC channels) without the if.

Reply:
I will remove the if condition in `done()`.